src/log.js

```diff
@@ -1,153 +1,389 @@
+    if (!log) throw new Error('Log instance not defined')
+    if (log.items.length < 1) throw new Error(`Can't serialize an empty log`)
+    return ipfs.object.put(log.toBuffer())
+      .then((res) => res.toJSON().multihash)
+  }
 
-        return Promise.map(entry.next, (f) => this._fetchRecursive(ipfs, f, all, amount, depth), { concurrency: 1 })
-          .then((res) => flatten(res.concat(result)))
+  /**
+   * Add an entry to a log
+   * @description Adds an entry to the Log and returns a new Log. Doesn't modify the original Log.
+   * @memberof Log
+   * @static
+   * @param {IPFS} ipfs An IPFS instance
+   * @param {Log} log - The Log to add the entry to
+   * @param {string|Buffer|Object|Array} data - Data of the entry to be added
+   * @returns {Log}
+   */
+  static append(ipfs, log, data) {
+    if (!ipfs) throw new Error('Ipfs instance not defined')
+    if (!log) throw new Error('Log instance not defined')
+    // Create the entry
+    return Entry.create(ipfs, data, log.heads)
+      .then((entry) => {
+        // Add the entry to the previous log entries
+        const items = log.items.slice().concat([entry])
+        // Set the heads of this log to the latest entry
+        const heads = [entry.hash]
+        // Create a new log instance
+        return new Log(ipfs, items, heads)
       })
   }
 
-  static getIpfsHash(ipfs, log) {
-    if (!ipfs) throw new Error("Ipfs instance not defined")
-    const data = new Buffer(JSON.stringify(log.snapshot))
-    return ipfs.object.put(data)
-      .then((res) => res.toJSON().multihash)
+  /**
+   * Join two logs
+   * 
+   * @description Joins two logs returning a new log. Doesn't mutate the original logs.
+   *
+   * @param {IPFS} [ipfs] An IPFS instance
+   * @param {Log} a
+   * @param {Log} b
+   *
+   * @example
+   * const log = Log.join(ipfs, log1, log2)
+   * 
+   * @returns {Log}
+   */
+  static join(ipfs, a, b, size) {
+    if (!ipfs) throw new Error('Ipfs instance not defined')
+    if (!a || !b) throw new Error('Log instance not defined')
+    if (!a.items || !b.items) throw new Error('Log to join must be an instance of Log')
+
+    // If size is not specified, join all entries by default
+    size = size ? size : a.items.length + b.items.length
+
+    // Get the heads from both logs and sort them by their IDs
+    const getHeadEntries = (log) => {
+      return log.heads
+      .map((e) => log.get(e))
+      .filter((e) => e !== undefined)
+      .slice()
+    }
+
+    const headsA = getHeadEntries(a)
+    const headsB = getHeadEntries(b)
+    const heads = headsA.concat(headsB)
+      .map((e) => e.hash)
+      .sort()
+
+    // Sort which log should come first based on heads' IDs
+    const aa = headsA[0] ? headsA[0].hash : null
+    const bb = headsB[0] ? headsB[0].hash : null
+    const isFirst = aa < bb
+    const log1 = isFirst ? a : b
+    const log2 = isFirst ? b : a
+
+    // Cap the size of the entries
+    const newEntries = log2.items.slice(0, size)
```

Comment:
Not sure what you mean exactly. In any case, we need to create a copy here in order to not mutate the previous items/array.